### PR TITLE
pngloss: init at unstable-2020-11-25

### PR DESCRIPTION
**Pull request by 2gn**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/tools/graphics/pngloss/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ];
+  builder = ./builder.sh;
+  
+  meta = with lib;{
+    description = "Lossy compression of PNG images";
+    homepage = "https://github.com/foobaz/pngloss";
+    license = licenses.mit;
+    platforms = platforms.all;
+  };
+}
```
[the remark this answers is not in the document]
Comment (2gn) [reply pkgs/tools/graphics/pngloss/default.nix:22]: Thank you for your review. To make this PR, I referred to some unofficial manuals that explains how to *build a project with Makefile using nix, so I didn't understand much of the philosophy.
> When you are fetching code from github, it is preferred to use fetchFromGitHub instead of fetchzip. It has an easier and more native interface.

I see. It looks far neater than the way I did. 

> The programs/tools go into nativeBuildInputs, because when you cross compile they still have to be native to your own architecture, not the one you're compiling to.

pkg-config was not installed on my darwin environment, and I didn't know where to put that in. Now I understand it...

>Then a bigger mistake, instead of supplying your own build script mkDerivation can do all of it for you already.
In your build script you still seem to copy the binary over manually, so I included that in the idiomatic way by using the installPhase.

I also didn't know that mkDerivation does all the stuff automagically...

I've learned a lot of things from this review. Again, thank you so much.

File: pkgs/tools/graphics/pngloss/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Lossy compression of PNG images";
+    homepage = "https://github.com/foobaz/pngloss";
+    license = licenses.mit;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ _2gn ];
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/tools/graphics/pngloss/default.nix:35]: Put a blank like here in the end.

And learn to use a power tool to fix these annoyances with whitespaces.

File: pkgs/tools/graphics/pngloss/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, pkg-config, libpng, zlib }:
+
+stdenv.mkDerivation rec {
+  pname = "pngloss";
+  version = "unstable-25-11-2020";
```
Comment (AndersonTorres) [inline pkgs/tools/graphics/pngloss/default.nix:5]:
```suggestion
  version = "unstable-2020-11-25";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
After this, squash all the fixups in one single commit.

**Comment by legendofmiracles** [comment]
Please also squash all package related commits

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -97,6 +97,12 @@
     github = "1000teslas";
     githubId = 47207223;
   };
+  _2gn = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:100]: This file should be in a separate commit, with the title

`maintainers: add myself`

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:100]: Also, put `pngloss: init at unstable-2020-11-25` on the other commit (removing everything else).

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (legendofmiracles) [reply maintainers/maintainer-list.nix:100]: > This file should be in a separate commit, with the title
> 
> `maintainers: add myself`

My guess is they misunderstood my squash commits, because that commit used to exist

Comment (2gn) [reply maintainers/maintainer-list.nix:100]: I've splitted squashed commit into "maintainer: add myself" and "pngloss: init at unstable-2020-11-25". Was it correct?

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:100]: Yes. The idea is two commits, one for the new maintainer and other for the package itself.

Comment (2gn) [reply maintainers/maintainer-list.nix:100]: Thank you! I've learned a lot from this pr.
